src/Commands/SixMans.py

```diff
@@ -286,134 +351,206 @@ def brokenQueue(player: Member) -> Embed:
     )
 
 
-def leaderboard(author: Member, mentions: List[Member], lbChannelId: int, *arg) -> Embed:
+def leaderboard(author: Member, mentions: str, lbChannelId: int, *arg) -> Embed:
     """
         Shows the leaderboard. Shows top 5 if no one mentioned. Shows player stats if mentioned or used keyword "me".
 
         Parameters:
             author: discord.Member - The author of the message
-            mentions: List[discord.Member] - The mentions in the message.
+            mentions: str - The content in the message.
             *arg - The rest of the args in the message.
 
         Returns:
             dicord.Embed - The embedded message to respond with.
     """
-    playerMentioned: bool = len(mentions) == 1
-    selfRank: bool = len(arg) == 1 and arg[0] == "me"
+    playerMentioned: bool = False
+    selfRank: bool = False
+    if (len(arg) > 0):
+        if ("<@!" in arg[0]):
+            split = mentions.split("<@!")
+            player_id = split[1][:-1]
+            if (player_id.isdigit()):
+                playerMentioned = True
+        elif (arg[0] == "me"):
+            selfRank = True
+
+    blueTeam, orangeTeam = Queue.getTeamList()
+    blueCap, orangeCap = Queue.getCaptains()
 
     if (playerMentioned or selfRank):
-
-        player = mentions[0] if playerMentioned else author
-
+        player = player_id if playerMentioned else str(author.id)
+        member = DataFiles.leaderboard.get(doc_id=int(player))
```

Comment:
I think it would be better to create a new function (if it doesn't already exist) in `Leaderboard.py` that returns a BallChaser given the ID. Queue and Leaderboard are really the only two files that should access the data files directly.

Reply:
Resolved.